lib/host/wasi/inode-linux.cpp

```diff
@@ -549,6 +549,10 @@ WasiExpect<void> INode::fdSeek(__wasi_filedelta_t Offset,
 }
 
 WasiExpect<void> INode::fdSync() const noexcept {
+  struct stat fdStat;
+  if (fstat(Fd, &fdStat) == 0 && S_ISDIR(fdStat.st_mode)) {
+    return {};
+  }
```

Comment:
This logic is: if we get a directory, we ignore it. I don't think this is expected behaviour.